functions/public/Invoke-WPFGetIso.ps1

```diff
@@ -260,19 +260,22 @@ function Invoke-WPFGetIso {
             $wimFile = $wimFile.Replace(".wim", ".esd").Trim()
         }
         $sync.MicrowinWindowsFlavors.Items.Clear()
-        Get-WindowsImage -ImagePath $wimFile | ForEach-Object {
+
+        $images = Get-WindowsImage -ImagePath $wimFile
+
+        # Populate the list of Windows flavors and find the Pro edition
+        $images | ForEach-Object {
             $imageIdx = $_.ImageIndex
             $imageName = $_.ImageName
+
             $sync.MicrowinWindowsFlavors.Items.Add("$imageIdx : $imageName")
-        }
-        $sync.MicrowinWindowsFlavors.SelectedIndex = 0
-        Write-Host "Finding suitable Pro edition. This can take some time. Do note that this is an automatic process that might not select the edition you want."
-        Get-WindowsImage -ImagePath $wimFile | ForEach-Object {
+
```

Comment:
```suggestion

        $sync.MicrowinWindowsFlavors.SelectedIndex = 0
        Write-Host "Finding suitable Pro edition. This can take some time. Do note that this is an automatic process that might not select the edition you want."
        
```

We need the end-user to see what's going on, don't you think?